packages/design-system-scripts/gulp/common/replaceTemplateTags.js

```diff
@@ -2,28 +2,31 @@
  * Replace template tags with string values
  */
 function replaceTemplateTags(str, options) {
+  // KSS processes the markdown in documentation content before we can replace the tags
+  // So unfortunately we need to match for the encoded version of the template tags
+  // When KSS is removed we can update the regex here
   if (options.rootPath) {
-    str = str.replace(/{{root}}/g, `/${options.rootPath}`);
+    str = str.replace(/{{root}}|%7B%7Broot%7D%7D/g, `/${options.rootPath}`);
```

Comment:
This isnt great and mainly has to do with how KSS works. Eventually we can get rid of this, and maybe even one day move to an implementation where we dont need to use `{{root}}` in the docs at all, and we can handle it for them.

Reply:
Would it make sense to pull these regex literal out into a constants? Or maybe we can put the branching in ternaries so the messy encoded strings aren't duplicated, like:
```js
return str
  .replace(/{{root}}|%7B%7Broot%7D%7D/g, options.rootPath ? `/${options.rootPath}` : '')
  .replace(/{{npm}}|%7B%7Bnpm%7D%7D/g, options.npmPackage ? `/${options.npmPackage}` : '')
  .replace(/{{github}}|%7B%7Bgithub%7D%7D/g, options.githubUrl ? `/${options.githubUrl}` : '')
  .replace(/{{name}}|%7B%7Bname%7D%7D/g, options.name ? `/${options.name}` : '')
```
or
```js
const tagRegex = tag => new RegExp(`${tag}|${encodeURI(tag)}`, 'g');
return str
  .replace(tagRegex('{{root}}'), options.rootPath ? `/${options.rootPath}` : '')
  .replace(tagRegex('{{npm}}'), options.npmPackage ? `/${options.npmPackage}` : '')
  .replace(tagRegex('{{github}}'), options.githubUrl ? `/${options.githubUrl}` : '')
  .replace(tagRegex('{{name}}'), options.name ? `/${options.name}` : '')
```
Or even make a singular `replaceTemplateTag` function that takes the string, the tag string, and the option

Reply:
I opted for the second option, much cleaner! I didn't make a singular `replaceTemplateTag` because  `rootPath` replaces slightly differently than that rest, it prepends a `/`. I could make the default `rootPath` `/` instead of empty string and make this nicer, but I didn't want to have to update that everywhere else for this PR.